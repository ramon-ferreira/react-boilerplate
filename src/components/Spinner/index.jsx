import { PropTypes } from 'prop-types';

import * as style from './style.scss';
import reactIconUrl from '@assets/reactjs-icon.svg';

function Spinner ({ alt = 'ReactJS Spinner' }) {
  return (<img className={style.image} src={reactIconUrl} alt={alt}/>);
}

Spinner.propTypes = {
  alt: PropTypes.string
};

export default Spinner;
