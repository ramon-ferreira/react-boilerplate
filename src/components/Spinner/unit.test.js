import renderer from 'react-test-renderer';
import Spinner from './index.jsx';

test('Component render', () => {
  const component = renderer.create(
    <Spinner></Spinner>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

/** Unit test example */
