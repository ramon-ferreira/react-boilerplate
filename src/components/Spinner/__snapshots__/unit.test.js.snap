// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component render 1`] = `
<img
  alt="ReactJS Spinner"
  className="image"
  src="test-file-stub"
/>
`;
